docroot/themes/humsci/humsci_basic/templates/menus/menu--main.html.twig

```diff
@@ -28,8 +28,47 @@
   {% if items is iterable %}
     {{ menus.nav_menu(items, 1, 'hb-main-nav') }}
   {% else %}
-    {# If custom markup is provided, emit it as is #}
+    {# If custom markup is provided, emit it as-is #}
     {{ items }}
   {% endif %}
   {% endspaceless %}
 </nav>
+
+{#
+  All menus in the main nav are expanded by default, which allows navigation if JavaScript is
+  disabled. If JavaScript is enabled, the code below will collapse the extraneous navigation.
+  We run this inline to ensure it runs immediately after the elements are added to the DOM,
+  preventing unsightly flashing on the page.
+#}
+<script>
```

Comment:
remove scripts from twig. this should live in its own js file.

Reply:
I hear what you're saying. Originally I had this script in it's own JS file, and was planning to a attach it to the menu with using a preprocess hook or twig's `attach_library()`. However, those techniques didn't give us enough control over load order to solve this specific problem.

For some background on the issue, we intentionally load our menus as expanded by default since it allows users to navigate when JS is disabled on their browser. However, this means that while the page is loading, the menus render as expanded before collapsing. H&S found this distracting and it was particularly bad on pages with lots of content, like the courses page with extra-long tables.

This inline script collapses the menus, and it's very sensitive to load order. If we load it in the header, there's nothing to collapse because menu DOM hasn't been constructed yet. If we load it near the footer, we run into the flashing issue because we have to wait for the full DOM to be rendered, plus any JS that must be downloaded and executed before it.

We've extracted all of the non-load-order-dependent menu javascript into their own JS files. The only part we're putting inline is the part that's required to collapse the menu, and we've commented the code to explain why it's needed. I feel comfortable moving forward with this exception, but if you know of a better way to accomplish this goal, then I'm open to it.

Reply:
this is a scenario my team is very unlikely to budge on. this needs to be extracted into a javascript file and loaded without `attach_library`. we dont allow those functions in twig templates. proper javascript file loaded into the header should be able act upon appropriate context when its available.

Reply:
also maybe some css could help with the flash. something like 
```
.js .thing ( visibility: hidden }
```
or whatever the use case is

Reply:
We can't rely on the `.js` class alone since the menus still need to be expandable. We may be able to come up with another class for representing load-progress though... I'm looking into this now. 👍

Reply:
Ok @pookmish, I just pushed up 13b74d8, which uses another approach to address this. Let me know if that looks good to you. 👍